src/FormixStore.ts

```diff
@@ -68,49 +69,49 @@ export default class FormixStore<T extends object> {
     this.validate();
   }
 
-  // TODO: improve types
-  setFieldValue(name: string, value: any) {
+  setFieldValue<Value>(name: string, value: Value) {
     set(this._values, name, value);
     this.validate();
   }
 
-  getFieldProps(name: string) {
+  getFieldProps<Value, Element extends HTMLInputElement>(
+    name: string
+  ): FieldProps<Value, Element> {
     const field = {
       name,
-      value: this.getValue(name),
+      value: this.getValue<Value>(name),
       onChange: this.handleChange,
       onBlur: this.handleBlur,
     };
 
     return field;
   }
 
-  getFieldMeta(name: string) {
+  getFieldMeta<Value>(name: string): FieldMeta<Value> {
     const meta = {
-      initialValue: this.getInitialValue(name),
-      value: this.getValue(name),
+      initialValue: this.getInitialValue<Value>(name),
+      value: this.getValue<Value>(name),
       error: this.getError(name),
       touched: this.getTouched(name),
     };
 
     return meta;
   }
 
-  getFieldHelpers(name: string) {
-    // TODO: improve types
+  getFieldHelpers<Value>(name: string): FieldHelpers<Value> {
     const helpers = {
-      setValue: (value: any) => this.setFieldValue(name, value),
+      setValue: (value: Value) => this.setFieldValue<Value>(name, value),
     };
 
     return helpers;
   }
 
-  getValue(name: string) {
-    return get(this._values, name);
+  getValue<Value>(name: string) {
+    return get(this._values, name) as Value;
```

Comment:
for all the get* functions, i think tou should define and follow one style: explicitly mark the function's return type or using "as type"

Reply:
done :)